{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>run_accession</th>\n",
       "      <th>SAMPLE_ID</th>\n",
       "      <th>TARGET</th>\n",
       "      <th>CELL_TYPE</th>\n",
       "      <th>REP</th>\n",
       "      <th>OLD_SAMPLE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR900927</td>\n",
       "      <td>RNA_ESC_1_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>ESC</td>\n",
       "      <td>1</td>\n",
       "      <td>RNA_ESC_Rep_1_JW-H9B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR900928</td>\n",
       "      <td>RNA_ESC_2_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>ESC</td>\n",
       "      <td>2</td>\n",
       "      <td>RNA_ESC_Rep_2_JW-H10B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR900929</td>\n",
       "      <td>RNA_MES_1_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>MES</td>\n",
       "      <td>1</td>\n",
       "      <td>RNA_MES_Rep_1_JW-H3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR900930</td>\n",
       "      <td>RNA_MES_2_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>MES</td>\n",
       "      <td>2</td>\n",
       "      <td>RNA_MES_Rep_2_JW-H4B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR900931</td>\n",
       "      <td>RNA_CP_1_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CP</td>\n",
       "      <td>1</td>\n",
       "      <td>RNA_CP_Rep_1_JW-H5B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRR900932</td>\n",
       "      <td>RNA_CP_2_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CP</td>\n",
       "      <td>2</td>\n",
       "      <td>RNA_CP_Rep_2_JW-H6B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRR900933</td>\n",
       "      <td>RNA_CM_1_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CM</td>\n",
       "      <td>1</td>\n",
       "      <td>RNA_CM_Rep_1_JW-H7B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRR900934</td>\n",
       "      <td>RNA_CM_2_Wamstad_2013_PE</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CM</td>\n",
       "      <td>2</td>\n",
       "      <td>RNA_CM_Rep_2_JW-H8B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  run_accession                  SAMPLE_ID TARGET CELL_TYPE  REP  \\\n",
       "0     SRR900927  RNA_ESC_1_Wamstad_2013_PE    RNA       ESC    1   \n",
       "1     SRR900928  RNA_ESC_2_Wamstad_2013_PE    RNA       ESC    2   \n",
       "2     SRR900929  RNA_MES_1_Wamstad_2013_PE    RNA       MES    1   \n",
       "3     SRR900930  RNA_MES_2_Wamstad_2013_PE    RNA       MES    2   \n",
       "4     SRR900931   RNA_CP_1_Wamstad_2013_PE    RNA        CP    1   \n",
       "5     SRR900932   RNA_CP_2_Wamstad_2013_PE    RNA        CP    2   \n",
       "6     SRR900933   RNA_CM_1_Wamstad_2013_PE    RNA        CM    1   \n",
       "7     SRR900934   RNA_CM_2_Wamstad_2013_PE    RNA        CM    2   \n",
       "\n",
       "           OLD_SAMPLE_ID  \n",
       "0   RNA_ESC_Rep_1_JW-H9B  \n",
       "1  RNA_ESC_Rep_2_JW-H10B  \n",
       "2   RNA_MES_Rep_1_JW-H3B  \n",
       "3   RNA_MES_Rep_2_JW-H4B  \n",
       "4    RNA_CP_Rep_1_JW-H5B  \n",
       "5    RNA_CP_Rep_2_JW-H6B  \n",
       "6    RNA_CM_Rep_1_JW-H7B  \n",
       "7    RNA_CM_Rep_2_JW-H8B  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "RNA_TABLE = pd.read_csv('./data/my_RNA_SRA_table.csv')\n",
    "RNA_TABLE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download SRA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import subprocess \n",
    "import shlex\n",
    "OUTDIR= '/usr/local/molbio/01_raw_data/Wamstad/'\n",
    "NT = '12'\n",
    "RANGE=(1,8)\n",
    "\n",
    "MAPFILES='/usr/local/molbio/02_map_files/mm10/'\n",
    "SSD_DIR='/Volumes/MariusSSD/prj/crg/02_map_files/mm10/Wamstad/'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "\n",
    "for i,row in RNA_TABLE.iterrows():\n",
    "    if i in range(RANGE[0],RANGE[1]):\n",
    "        SAMPLE_ID = row['SAMPLE_ID']\n",
    "        SRA = row['run_accession']\n",
    "        print(f'\\nIndex   = {i}')\n",
    "        print(f'SRA       = {SRA}')\n",
    "        print(f'SAMPLE_ID = {SAMPLE_ID}') \n",
    "        print(f'OUTDIR    = {OUTDIR}\\n')\n",
    "        ! parallel-fastq-dump --tmpdir . --sra-id {SRA} --threads {NT} --outdir {OUTDIR} --gzip --split-files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Map with TopHat2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Index     = 1\n",
      "SRA       = SRR900928\n",
      "SAMPLE_ID = RNA_ESC_2_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900928_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900928_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_ESC_2_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900928_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900928_2.fastq.gz -> $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900928_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900928_2.fastq.gz 2> info/tophat_RNA_ESC_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_ESC_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_ESC_2_Wamstad_2013_PE.txt\n",
      "\u001b[0m42882019 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "42882019 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "42882019 + 0 mapped (100.00% : N/A)\n",
      "42882019 + 0 primary mapped (100.00% : N/A)\n",
      "42882019 + 0 paired in sequencing\n",
      "21505064 + 0 read1\n",
      "21376955 + 0 read2\n",
      "30682322 + 0 properly paired (71.55% : N/A)\n",
      "41704510 + 0 with itself and mate mapped\n",
      "1177509 + 0 singletons (2.75% : N/A)\n",
      "2195944 + 0 with mate mapped to a different chr\n",
      "2195944 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam -> RNA_ESC_2_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_ESC_2_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_ESC_2_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_ESC_2_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.737  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 44.25  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 2655  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_ESC_2_Wamstad_2013_PE/\n",
      "RNA_ESC_2_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_ESC_2_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_ESC_2_Wamstad_2013_PE/deletions.bed\n",
      "RNA_ESC_2_Wamstad_2013_PE/insertions.bed\n",
      "RNA_ESC_2_Wamstad_2013_PE/junctions.bed\n",
      "RNA_ESC_2_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_ESC_2_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/run.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_ESC_2_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.38G bytes  received 714 bytes  212.67M bytes/sec\n",
      "total size is 1.38G  speedup is 1.00\n",
      "\n",
      "Index     = 2\n",
      "SRA       = SRR900929\n",
      "SAMPLE_ID = RNA_MES_1_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900929_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900929_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_MES_1_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900929_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900929_2.fastq.gz -> $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900929_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900929_2.fastq.gz 2> info/tophat_RNA_MES_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_MES_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_MES_1_Wamstad_2013_PE.txt\n",
      "\u001b[0m40642659 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "40642659 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "40642659 + 0 mapped (100.00% : N/A)\n",
      "40642659 + 0 primary mapped (100.00% : N/A)\n",
      "40642659 + 0 paired in sequencing\n",
      "20470741 + 0 read1\n",
      "20171918 + 0 read2\n",
      "29090072 + 0 properly paired (71.58% : N/A)\n",
      "39778854 + 0 with itself and mate mapped\n",
      "863805 + 0 singletons (2.13% : N/A)\n",
      "1274226 + 0 with mate mapped to a different chr\n",
      "1274226 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam -> RNA_MES_1_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_MES_1_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_MES_1_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_MES_1_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.704  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 42.25  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 2535  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_MES_1_Wamstad_2013_PE/\n",
      "RNA_MES_1_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_MES_1_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_MES_1_Wamstad_2013_PE/deletions.bed\n",
      "RNA_MES_1_Wamstad_2013_PE/insertions.bed\n",
      "RNA_MES_1_Wamstad_2013_PE/junctions.bed\n",
      "RNA_MES_1_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_MES_1_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/run.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_MES_1_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.27G bytes  received 714 bytes  231.80M bytes/sec\n",
      "total size is 1.27G  speedup is 1.00\n",
      "\n",
      "Index     = 3\n",
      "SRA       = SRR900930\n",
      "SAMPLE_ID = RNA_MES_2_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900930_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900930_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_MES_2_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900930_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900930_2.fastq.gz -> $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900930_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900930_2.fastq.gz 2> info/tophat_RNA_MES_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_MES_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_MES_2_Wamstad_2013_PE.txt\n",
      "\u001b[0m39378171 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "39378171 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "39378171 + 0 mapped (100.00% : N/A)\n",
      "39378171 + 0 primary mapped (100.00% : N/A)\n",
      "39378171 + 0 paired in sequencing\n",
      "19807014 + 0 read1\n",
      "19571157 + 0 read2\n",
      "28528438 + 0 properly paired (72.45% : N/A)\n",
      "38535112 + 0 with itself and mate mapped\n",
      "843059 + 0 singletons (2.14% : N/A)\n",
      "1398562 + 0 with mate mapped to a different chr\n",
      "1398562 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam -> RNA_MES_2_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_MES_2_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_MES_2_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_MES_2_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.699  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 41.983  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 2519  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_MES_2_Wamstad_2013_PE/\n",
      "RNA_MES_2_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_MES_2_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_MES_2_Wamstad_2013_PE/deletions.bed\n",
      "RNA_MES_2_Wamstad_2013_PE/insertions.bed\n",
      "RNA_MES_2_Wamstad_2013_PE/junctions.bed\n",
      "RNA_MES_2_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_MES_2_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/run.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_MES_2_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.25G bytes  received 714 bytes  226.79M bytes/sec\n",
      "total size is 1.25G  speedup is 1.00\n",
      "\n",
      "Index     = 4\n",
      "SRA       = SRR900931\n",
      "SAMPLE_ID = RNA_CP_1_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900931_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900931_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_CP_1_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900931_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900931_2.fastq.gz -> $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900931_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900931_2.fastq.gz 2> info/tophat_RNA_CP_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_CP_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_CP_1_Wamstad_2013_PE.txt\n",
      "\u001b[0m41585604 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "41585604 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "41585604 + 0 mapped (100.00% : N/A)\n",
      "41585604 + 0 primary mapped (100.00% : N/A)\n",
      "41585604 + 0 paired in sequencing\n",
      "20921436 + 0 read1\n",
      "20664168 + 0 read2\n",
      "29423734 + 0 properly paired (70.75% : N/A)\n",
      "40672510 + 0 with itself and mate mapped\n",
      "913094 + 0 singletons (2.20% : N/A)\n",
      "1460722 + 0 with mate mapped to a different chr\n",
      "1460722 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam -> RNA_CP_1_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_CP_1_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_CP_1_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_CP_1_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.74  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 44.4  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 2664  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_CP_1_Wamstad_2013_PE/\n",
      "RNA_CP_1_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_CP_1_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_CP_1_Wamstad_2013_PE/deletions.bed\n",
      "RNA_CP_1_Wamstad_2013_PE/insertions.bed\n",
      "RNA_CP_1_Wamstad_2013_PE/junctions.bed\n",
      "RNA_CP_1_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_CP_1_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/run.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_CP_1_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.34G bytes  received 714 bytes  206.68M bytes/sec\n",
      "total size is 1.34G  speedup is 1.00\n",
      "\n",
      "Index     = 5\n",
      "SRA       = SRR900932\n",
      "SAMPLE_ID = RNA_CP_2_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900932_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900932_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_CP_2_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900932_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900932_2.fastq.gz -> $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900932_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900932_2.fastq.gz 2> info/tophat_RNA_CP_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_CP_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_CP_2_Wamstad_2013_PE.txt\n",
      "\u001b[0m40731481 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "40731481 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "40731481 + 0 mapped (100.00% : N/A)\n",
      "40731481 + 0 primary mapped (100.00% : N/A)\n",
      "40731481 + 0 paired in sequencing\n",
      "20473358 + 0 read1\n",
      "20258123 + 0 read2\n",
      "28663514 + 0 properly paired (70.37% : N/A)\n",
      "39902200 + 0 with itself and mate mapped\n",
      "829281 + 0 singletons (2.04% : N/A)\n",
      "1337610 + 0 with mate mapped to a different chr\n",
      "1337610 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam -> RNA_CP_2_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_CP_2_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_CP_2_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_CP_2_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.726  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 43.566  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 2614  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_CP_2_Wamstad_2013_PE/\n",
      "RNA_CP_2_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_CP_2_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_CP_2_Wamstad_2013_PE/deletions.bed\n",
      "RNA_CP_2_Wamstad_2013_PE/insertions.bed\n",
      "RNA_CP_2_Wamstad_2013_PE/junctions.bed\n",
      "RNA_CP_2_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_CP_2_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/run.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_CP_2_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.30G bytes  received 714 bytes  200.39M bytes/sec\n",
      "total size is 1.30G  speedup is 1.00\n",
      "\n",
      "Index     = 6\n",
      "SRA       = SRR900933\n",
      "SAMPLE_ID = RNA_CM_1_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900933_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900933_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_CM_1_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900933_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900933_2.fastq.gz -> $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900933_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900933_2.fastq.gz 2> info/tophat_RNA_CM_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_CM_1_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_CM_1_Wamstad_2013_PE.txt\n",
      "\u001b[0m57779285 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "57779285 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "57779285 + 0 mapped (100.00% : N/A)\n",
      "57779285 + 0 primary mapped (100.00% : N/A)\n",
      "57779285 + 0 paired in sequencing\n",
      "28999838 + 0 read1\n",
      "28779447 + 0 read2\n",
      "44942906 + 0 properly paired (77.78% : N/A)\n",
      "56712564 + 0 with itself and mate mapped\n",
      "1066721 + 0 singletons (1.85% : N/A)\n",
      "1710366 + 0 with mate mapped to a different chr\n",
      "1710366 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam -> RNA_CM_1_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_CM_1_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_CM_1_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_CM_1_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.934  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 56.083  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 3365  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_CM_1_Wamstad_2013_PE/\n",
      "RNA_CM_1_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_CM_1_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_CM_1_Wamstad_2013_PE/deletions.bed\n",
      "RNA_CM_1_Wamstad_2013_PE/insertions.bed\n",
      "RNA_CM_1_Wamstad_2013_PE/junctions.bed\n",
      "RNA_CM_1_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_CM_1_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/run.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_CM_1_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.73G bytes  received 714 bytes  203.09M bytes/sec\n",
      "total size is 1.73G  speedup is 1.00\n",
      "\n",
      "Index     = 7\n",
      "SRA       = SRR900934\n",
      "SAMPLE_ID = RNA_CM_2_Wamstad_2013_PE\n",
      "FQ_DIR    = /usr/local/molbio/01_raw_data/Wamstad/\n",
      "FQ1 = /usr/local/molbio/01_raw_data/Wamstad/SRR900934_1.fastq.gz\n",
      "FQ2 = /usr/local/molbio/01_raw_data/Wamstad/SRR900934_2.fastq.gz\n",
      "OUTDIR = /usr/local/molbio/02_map_files/mm10/RNA_CM_2_Wamstad_2013_PE\n",
      "\n",
      "\u001b[1;32m%%%% ProcessRNAseq.pl by Enrique Blanco @ CRG (2018)\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 0.  Reading options\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Creating the info directory\n",
      "\u001b[0m\u001b[1;32m%%%% It is already existing\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 1.  Mapping the RNAseq sample/s (/usr/local/molbio/01_raw_data/Wamstad/SRR900934_1.fastq.gz,/usr/local/molbio/01_raw_data/Wamstad/SRR900934_2.fastq.gz -> $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running tophat2 --zpacker pigz --transcriptome-index=$MOUSE10/transcriptome_index/refGene --no-coverage-search --mate-inner-dist 176 --mate-std-dev 11 -p 3 -g 1 -o $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE --library-type=fr-firststrand $MOUSE10/genome /usr/local/molbio/01_raw_data/Wamstad/SRR900934_1.fastq.gz /usr/local/molbio/01_raw_data/Wamstad/SRR900934_2.fastq.gz 2> info/tophat_RNA_CM_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 2.  Obtaining the statistics ($MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools flagstat $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam > info/flagstat_RNA_CM_2_Wamstad_2013_PE.txt\u001b[0m\u001b[1;32m%%%% Running cat info/flagstat_RNA_CM_2_Wamstad_2013_PE.txt\n",
      "\u001b[0m55013670 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "55013670 + 0 primary\n",
      "0 + 0 secondary\n",
      "0 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "55013670 + 0 mapped (100.00% : N/A)\n",
      "55013670 + 0 primary mapped (100.00% : N/A)\n",
      "55013670 + 0 paired in sequencing\n",
      "27571852 + 0 read1\n",
      "27441818 + 0 read2\n",
      "40261700 + 0 properly paired (73.18% : N/A)\n",
      "53833448 + 0 with itself and mate mapped\n",
      "1180222 + 0 singletons (2.15% : N/A)\n",
      "1841430 + 0 with mate mapped to a different chr\n",
      "1841430 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Stage 3.  Building the normalized profile/s for UCSC ($MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam -> RNA_CM_2_Wamstad_2013_PE)\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -h -o $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.sam $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:+\" $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSplus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running grep \"XS:A:-\" $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.sam > $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSminus.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running samtools view -H $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam > $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/header.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSplus.sam > $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running cat $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/header.sam $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSminus.sam > $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSplusH.sam RNA_CM_2_Wamstad_2013_PE_plus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running buildChIPprofile -p $MOUSE10/ChromInfo.txt $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/accepted_hits_XSminusH.sam RNA_CM_2_Wamstad_2013_PE_minus\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Running rm -f $MAPFILES/mm10/RNA_CM_2_Wamstad_2013_PE/*.sam\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0m\u001b[1;32m%%%% Successful termination:\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (hours): 0.897  hours\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (minutes): 53.866  mins\n",
      "\u001b[0m\u001b[1;32m%%%% Total running time (seconds): 3232  secs\u001b[0m\u001b[1;32m [OK]\n",
      "\n",
      "\u001b[0mbuilding file list ... done\n",
      "RNA_CM_2_Wamstad_2013_PE/\n",
      "RNA_CM_2_Wamstad_2013_PE/accepted_hits.bam\n",
      "RNA_CM_2_Wamstad_2013_PE/align_summary.txt\n",
      "RNA_CM_2_Wamstad_2013_PE/deletions.bed\n",
      "RNA_CM_2_Wamstad_2013_PE/insertions.bed\n",
      "RNA_CM_2_Wamstad_2013_PE/junctions.bed\n",
      "RNA_CM_2_Wamstad_2013_PE/prep_reads.info\n",
      "RNA_CM_2_Wamstad_2013_PE/unmapped.bam\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bam_merge_um.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.left_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie.right_kept_reads.m2g_um_unmapped.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/bowtie_build.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/gtf_juncs.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/juncs_db.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/long_spanning_reads.segs.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.err\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/m2g_left_kept_reads.out\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.err\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/m2g_right_kept_reads.out\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/prep_reads.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/reports.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/reports.merge_bam.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/reports.samtools_sort.log0\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/reports.samtools_sort.log1\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/reports.samtools_sort.log2\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/run.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/segment_juncs.log\n",
      "RNA_CM_2_Wamstad_2013_PE/logs/tophat.log\n",
      "\n",
      "sent 1.70G bytes  received 714 bytes  200.09M bytes/sec\n",
      "total size is 1.70G  speedup is 1.00\n",
      "CPU times: user 1min 19s, sys: 24.9 s, total: 1min 44s\n",
      "Wall time: 5h 27min 12s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "for i,row in RNA_TABLE.iterrows():\n",
    "    if i in range(RANGE[0],RANGE[1]):\n",
    "        try:\n",
    "            SRA = row['run_accession']\n",
    "            SAMPLE_ID = row['SAMPLE_ID']\n",
    "            FQ1 = OUTDIR+SRA+'_1.fastq.gz'\n",
    "            FQ2 = OUTDIR+SRA+'_2.fastq.gz'\n",
    "            print(f'\\nIndex     = {i}')\n",
    "            print(f'SRA       = {SRA}')\n",
    "            print(f'SAMPLE_ID = {SAMPLE_ID}')\n",
    "            print(f'FQ_DIR    = {OUTDIR}')\n",
    "            print(f'FQ1 = {FQ1}')\n",
    "            print(f'FQ2 = {FQ2}')\n",
    "            print(f'OUTDIR = {MAPFILES}{SAMPLE_ID}\\n')\n",
    "            #subprocess.run(['pigz', '-d', '-k', '-p','6', FQ1 +'.gz'])\n",
    "            #subprocess.run(['pigz', '-d', '-k', '-p','6', FQ2 +'.gz'])\n",
    "            #subprocess.run(shlex.split(f'ProcessRNAseq.pl -v -p mouse mm10 {FQ1} {FQ2} {SAMPLE_ID}'))\n",
    "            ! ./ProcessRNAseq.pl -v -p mouse mm10 {FQ1} {FQ2} {SAMPLE_ID}\n",
    "            #subprocess.run(['rm', '-f', FQ1, FQ2])\n",
    "            ! rsync -avh --remove-source-files {MAPFILES}{SAMPLE_ID} {SSD_DIR}\n",
    "        except Exception:\n",
    "            print('Error')\n",
    "            break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Assert all the SAMPLE IDs have a BAM file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "SAMPLE_IDS = pd.Index([sample_id  for sample_id in RNA_TABLE['SAMPLE_ID']])\n",
    "BAMS = pd.Index(os.listdir(SSD_DIR))\n",
    "assert len(SAMPLE_IDS.intersection(BAMS)) == len(SAMPLE_IDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['RNA_ESC_1_Wamstad_2013_PE', 'RNA_ESC_2_Wamstad_2013_PE',\n",
       "       'RNA_MES_1_Wamstad_2013_PE', 'RNA_MES_2_Wamstad_2013_PE',\n",
       "       'RNA_CP_1_Wamstad_2013_PE', 'RNA_CP_2_Wamstad_2013_PE',\n",
       "       'RNA_CM_1_Wamstad_2013_PE', 'RNA_CM_2_Wamstad_2013_PE'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "SAMPLE_IDS.intersection(BAMS)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "DL2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
