{"cells":[{"cell_type":"code","execution_count":1,"metadata":{},"outputs":[],"source":["import pandas as pd\n","import matplotlib as mpl\n","import matplotlib.pyplot as plt\n","\n","import gseapy as gp\n","from gseapy import barplot, dotplot\n","import pickle        \n"]},{"cell_type":"markdown","metadata":{},"source":["## Gene-sets from DBs"]},{"cell_type":"code","execution_count":2,"metadata":{},"outputs":[{"data":{"text/plain":["['ARCHS4_Cell-lines',\n"," 'ARCHS4_IDG_Coexp',\n"," 'ARCHS4_Kinases_Coexp',\n"," 'ARCHS4_TFs_Coexp',\n"," 'ARCHS4_Tissues',\n"," 'Achilles_fitness_decrease',\n"," 'Achilles_fitness_increase',\n"," 'Aging_Perturbations_from_GEO_down',\n"," 'Aging_Perturbations_from_GEO_up',\n"," 'Allen_Brain_Atlas_10x_scRNA_2021',\n"," 'Allen_Brain_Atlas_down',\n"," 'Allen_Brain_Atlas_up',\n"," 'Azimuth_2023',\n"," 'Azimuth_Cell_Types_2021',\n"," 'BioCarta_2013',\n"," 'BioCarta_2015',\n"," 'BioCarta_2016',\n"," 'BioPlanet_2019',\n"," 'BioPlex_2017',\n"," 'CCLE_Proteomics_2020',\n"," 'CORUM',\n"," 'COVID-19_Related_Gene_Sets',\n"," 'COVID-19_Related_Gene_Sets_2021',\n"," 'Cancer_Cell_Line_Encyclopedia',\n"," 'CellMarker_2024',\n"," 'CellMarker_Augmented_2021',\n"," 'ChEA_2013',\n"," 'ChEA_2015',\n"," 'ChEA_2016',\n"," 'ChEA_2022',\n"," 'Chromosome_Location',\n"," 'Chromosome_Location_hg19',\n"," 'ClinVar_2019',\n"," 'DGIdb_Drug_Targets_2024',\n"," 'DSigDB',\n"," 'Data_Acquisition_Method_Most_Popular_Genes',\n"," 'DepMap_CRISPR_GeneDependency_CellLines_2023',\n"," 'DepMap_WG_CRISPR_Screens_Broad_CellLines_2019',\n"," 'DepMap_WG_CRISPR_Screens_Sanger_CellLines_2019',\n"," 'Descartes_Cell_Types_and_Tissue_2021',\n"," 'Diabetes_Perturbations_GEO_2022',\n"," 'DisGeNET',\n"," 'Disease_Perturbations_from_GEO_down',\n"," 'Disease_Perturbations_from_GEO_up',\n"," 'Disease_Signatures_from_GEO_down_2014',\n"," 'Disease_Signatures_from_GEO_up_2014',\n"," 'DrugMatrix',\n"," 'Drug_Perturbations_from_GEO_2014',\n"," 'Drug_Perturbations_from_GEO_down',\n"," 'Drug_Perturbations_from_GEO_up',\n"," 'ENCODE_Histone_Modifications_2013',\n"," 'ENCODE_Histone_Modifications_2015',\n"," 'ENCODE_TF_ChIP-seq_2014',\n"," 'ENCODE_TF_ChIP-seq_2015',\n"," 'ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X',\n"," 'ESCAPE',\n"," 'Elsevier_Pathway_Collection',\n"," 'Enrichr_Libraries_Most_Popular_Genes',\n"," 'Enrichr_Submissions_TF-Gene_Coocurrence',\n"," 'Enrichr_Users_Contributed_Lists_2020',\n"," 'Epigenomics_Roadmap_HM_ChIP-seq',\n"," 'FANTOM6_lncRNA_KD_DEGs',\n"," 'GO_Biological_Process_2013',\n"," 'GO_Biological_Process_2015',\n"," 'GO_Biological_Process_2017',\n"," 'GO_Biological_Process_2017b',\n"," 'GO_Biological_Process_2018',\n"," 'GO_Biological_Process_2021',\n"," 'GO_Biological_Process_2023',\n"," 'GO_Cellular_Component_2013',\n"," 'GO_Cellular_Component_2015',\n"," 'GO_Cellular_Component_2017',\n"," 'GO_Cellular_Component_2017b',\n"," 'GO_Cellular_Component_2018',\n"," 'GO_Cellular_Component_2021',\n"," 'GO_Cellular_Component_2023',\n"," 'GO_Molecular_Function_2013',\n"," 'GO_Molecular_Function_2015',\n"," 'GO_Molecular_Function_2017',\n"," 'GO_Molecular_Function_2017b',\n"," 'GO_Molecular_Function_2018',\n"," 'GO_Molecular_Function_2021',\n"," 'GO_Molecular_Function_2023',\n"," 'GTEx_Aging_Signatures_2021',\n"," 'GTEx_Tissue_Expression_Down',\n"," 'GTEx_Tissue_Expression_Up',\n"," 'GTEx_Tissues_V8_2023',\n"," 'GWAS_Catalog_2019',\n"," 'GWAS_Catalog_2023',\n"," 'GeDiPNet_2023',\n"," 'GeneSigDB',\n"," 'Gene_Perturbations_from_GEO_down',\n"," 'Gene_Perturbations_from_GEO_up',\n"," 'Genes_Associated_with_NIH_Grants',\n"," 'Genome_Browser_PWMs',\n"," 'GlyGen_Glycosylated_Proteins_2022',\n"," 'HDSigDB_Human_2021',\n"," 'HDSigDB_Mouse_2021',\n"," 'HMDB_Metabolites',\n"," 'HMS_LINCS_KinomeScan',\n"," 'HomoloGene',\n"," 'HuBMAP_ASCT_plus_B_augmented_w_RNAseq_Coexpression',\n"," 'HuBMAP_ASCTplusB_augmented_2022',\n"," 'HumanCyc_2015',\n"," 'HumanCyc_2016',\n"," 'Human_Gene_Atlas',\n"," 'Human_Phenotype_Ontology',\n"," 'IDG_Drug_Targets_2022',\n"," 'InterPro_Domains_2019',\n"," 'Jensen_COMPARTMENTS',\n"," 'Jensen_DISEASES',\n"," 'Jensen_TISSUES',\n"," 'KEA_2013',\n"," 'KEA_2015',\n"," 'KEGG_2013',\n"," 'KEGG_2015',\n"," 'KEGG_2016',\n"," 'KEGG_2019_Human',\n"," 'KEGG_2019_Mouse',\n"," 'KEGG_2021_Human',\n"," 'KOMP2_Mouse_Phenotypes_2022',\n"," 'Kinase_Perturbations_from_GEO_down',\n"," 'Kinase_Perturbations_from_GEO_up',\n"," 'L1000_Kinase_and_GPCR_Perturbations_down',\n"," 'L1000_Kinase_and_GPCR_Perturbations_up',\n"," 'LINCS_L1000_CRISPR_KO_Consensus_Sigs',\n"," 'LINCS_L1000_Chem_Pert_Consensus_Sigs',\n"," 'LINCS_L1000_Chem_Pert_down',\n"," 'LINCS_L1000_Chem_Pert_up',\n"," 'LINCS_L1000_Ligand_Perturbations_down',\n"," 'LINCS_L1000_Ligand_Perturbations_up',\n"," 'Ligand_Perturbations_from_GEO_down',\n"," 'Ligand_Perturbations_from_GEO_up',\n"," 'MAGMA_Drugs_and_Diseases',\n"," 'MAGNET_2023',\n"," 'MCF7_Perturbations_from_GEO_down',\n"," 'MCF7_Perturbations_from_GEO_up',\n"," 'MGI_Mammalian_Phenotype_2013',\n"," 'MGI_Mammalian_Phenotype_2017',\n"," 'MGI_Mammalian_Phenotype_Level_3',\n"," 'MGI_Mammalian_Phenotype_Level_4',\n"," 'MGI_Mammalian_Phenotype_Level_4_2019',\n"," 'MGI_Mammalian_Phenotype_Level_4_2021',\n"," 'MGI_Mammalian_Phenotype_Level_4_2024',\n"," 'MSigDB_Computational',\n"," 'MSigDB_Hallmark_2020',\n"," 'MSigDB_Oncogenic_Signatures',\n"," 'Metabolomics_Workbench_Metabolites_2022',\n"," 'Microbe_Perturbations_from_GEO_down',\n"," 'Microbe_Perturbations_from_GEO_up',\n"," 'MoTrPAC_2023',\n"," 'Mouse_Gene_Atlas',\n"," 'NCI-60_Cancer_Cell_Lines',\n"," 'NCI-Nature_2016',\n"," 'NIH_Funded_PIs_2017_AutoRIF_ARCHS4_Predictions',\n"," 'NIH_Funded_PIs_2017_GeneRIF_ARCHS4_Predictions',\n"," 'NIH_Funded_PIs_2017_Human_AutoRIF',\n"," 'NIH_Funded_PIs_2017_Human_GeneRIF',\n"," 'NURSA_Human_Endogenous_Complexome',\n"," 'OMIM_Disease',\n"," 'OMIM_Expanded',\n"," 'Old_CMAP_down',\n"," 'Old_CMAP_up',\n"," 'Orphanet_Augmented_2021',\n"," 'PFOCR_Pathways',\n"," 'PFOCR_Pathways_2023',\n"," 'PPI_Hub_Proteins',\n"," 'PanglaoDB_Augmented_2021',\n"," 'Panther_2015',\n"," 'Panther_2016',\n"," 'Pfam_Domains_2019',\n"," 'Pfam_InterPro_Domains',\n"," 'PheWeb_2019',\n"," 'PhenGenI_Association_2021',\n"," 'Phosphatase_Substrates_from_DEPOD',\n"," 'ProteomicsDB_2020',\n"," 'Proteomics_Drug_Atlas_2023',\n"," 'RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO',\n"," 'RNAseq_Automatic_GEO_Signatures_Human_Down',\n"," 'RNAseq_Automatic_GEO_Signatures_Human_Up',\n"," 'RNAseq_Automatic_GEO_Signatures_Mouse_Down',\n"," 'RNAseq_Automatic_GEO_Signatures_Mouse_Up',\n"," 'Rare_Diseases_AutoRIF_ARCHS4_Predictions',\n"," 'Rare_Diseases_AutoRIF_Gene_Lists',\n"," 'Rare_Diseases_GeneRIF_ARCHS4_Predictions',\n"," 'Rare_Diseases_GeneRIF_Gene_Lists',\n"," 'Reactome_2013',\n"," 'Reactome_2015',\n"," 'Reactome_2016',\n"," 'Reactome_2022',\n"," 'Rummagene_kinases',\n"," 'Rummagene_signatures',\n"," 'Rummagene_transcription_factors',\n"," 'SILAC_Phosphoproteomics',\n"," 'SubCell_BarCode',\n"," 'SynGO_2022',\n"," 'SynGO_2024',\n"," 'SysMyo_Muscle_Gene_Sets',\n"," 'TF-LOF_Expression_from_GEO',\n"," 'TF_Perturbations_Followed_by_Expression',\n"," 'TG_GATES_2020',\n"," 'TRANSFAC_and_JASPAR_PWMs',\n"," 'TRRUST_Transcription_Factors_2019',\n"," 'Table_Mining_of_CRISPR_Studies',\n"," 'Tabula_Muris',\n"," 'Tabula_Sapiens',\n"," 'TargetScan_microRNA',\n"," 'TargetScan_microRNA_2017',\n"," 'The_Kinase_Library_2023',\n"," 'The_Kinase_Library_2024',\n"," 'Tissue_Protein_Expression_from_Human_Proteome_Map',\n"," 'Tissue_Protein_Expression_from_ProteomicsDB',\n"," 'Transcription_Factor_PPIs',\n"," 'UK_Biobank_GWAS_v1',\n"," 'Virus-Host_PPI_P-HIPSTer_2020',\n"," 'VirusMINT',\n"," 'Virus_Perturbations_from_GEO_down',\n"," 'Virus_Perturbations_from_GEO_up',\n"," 'WikiPathway_2021_Human',\n"," 'WikiPathway_2023_Human',\n"," 'WikiPathways_2013',\n"," 'WikiPathways_2015',\n"," 'WikiPathways_2016',\n"," 'WikiPathways_2019_Human',\n"," 'WikiPathways_2019_Mouse',\n"," 'WikiPathways_2024_Human',\n"," 'WikiPathways_2024_Mouse',\n"," 'dbGaP',\n"," 'huMAP',\n"," 'lncHUB_lncRNA_Co-Expression',\n"," 'miRTarBase_2017']"]},"execution_count":2,"metadata":{},"output_type":"execute_result"}],"source":["GENE_SETS = {}\n","MOUSE_DB = gp.get_library_name(organism='Mouse')\n","MOUSE_DB"]},{"cell_type":"markdown","metadata":{},"source":["### WikiPathways2019_Mouse: PluriNetWork_WP1763"]},{"cell_type":"code","execution_count":3,"metadata":{},"outputs":[],"source":["wp = gp.get_library(name='WikiPathways_2019_Mouse',organism='Mouse')"]},{"cell_type":"code","execution_count":4,"metadata":{},"outputs":[],"source":["ESC_PATHWAYS= [  \n","                'Hedgehog Signaling Pathway WP116',\n","                'PluriNetWork WP1763',\n","                'ESC Pluripotency Pathways WP339',\n","                'Wnt Signaling Pathway WP403',]\n","CARDIO_PATHWAYS = ['Striated Muscle Contraction WP216', \n","                'Heart Development WP2067',\n","                'Delta-Notch Signaling Pathway WP265',]\n","\n","MES_PATHWAYS = ['Epithelial Mesenchymal Transition WP306',]"]},{"cell_type":"markdown","metadata":{},"source":["### Add Early Mesodermal lineage markers \n","https://www.rndsystems.com/research-area/early-mesodermal-lineage-markers"]},{"cell_type":"code","execution_count":5,"metadata":{},"outputs":[],"source":["MES = [\n","    'Cdh2',          # N-Cadherin\n","\n","    # Early Mesodermal Cell Secreted Factors\n","    'Inhba',         # Activin A\n","    'Inhba',         # Activin AB (component 1)\n","    'Inhbb',         # Activin AB (component 2)\n","    'Inhba',         # Activin AC (component 1)\n","    'Inhbc',         # Activin AC (component 2)\n","    'Inhbb',         # Activin B\n","    'Inhbc',         # Activin C\n","\n","    # BMP and Other Activin Receptor Activators/Inhibitors\n","    'Bmp2',          # BMP-2\n","    'Bmp2',          # BMP-2/BMP-4 (component 1)\n","    'Bmp4',          # BMP-2/BMP-4 (component 2)\n","    'Bmp2',          # BMP-2/BMP-6 Heterodimer (component 1)\n","    'Bmp6',          # BMP-2/BMP-6 Heterodimer (component 2)\n","    'Bmp2',          # BMP-2/BMP-7 Heterodimer (component 1)\n","    'Bmp7',          # BMP-2/BMP-7 Heterodimer (component 2)\n","    'Bmp2',          # BMP-2a (no specific `Bmp2a` in mouse)\n","    'Bmp4',          # BMP-4\n","    'Bmp6',          # BMP-6\n","    'Bmp7',          # BMP-7\n","    'Cfc1',          # Cryptic\n","    'Fabp4',         # FABP4/A-FABP\n","    'Fgf5',          # FGF-5\n","    'Gdf1',          # GDF-1\n","    'Gdf3',          # GDF-3\n","    'Inhba',         # INHBA\n","    'Inhbb',         # INHBB\n","    'Inhbe',         # INHBE\n","    'Nodal',         # Nodal\n","    'Tgfb1',         # TGF-beta 1\n","    'Tgfb2',         # TGF-beta 2\n","    'Tgfb3',         # TGF-beta 3\n","    'Wnt3a',         # Wnt-3a\n","    'Wnt8a',         # Wnt-8a\n","    'Wnt3',          # Wnt-3\n","\n","    # ER Proteins in Early Mesodermal Cells\n","    'Mesd',        # MESDC2\n","    'Ncln',          # Nicalin\n","\n","    # Transcription Factors in Early Mesodermal Cells\n","    'T',             # Brachyury\n","    'Eomes',         # EOMES\n","    'Foxc1',         # FoxC1\n","    'Foxf1',         # FoxF1\n","    'Gsc',           # Goosecoid\n","    'Hand1',         # HAND1\n","    'Mixl1',         # MIXL1\n","    'Snai2',         # Slug\n","    'Snai1',         # Snail\n","    'Tbx6',          # TBX6\n","    'Twist1',        # Twist-1\n","    'Twist2'         # Twist-2\n","]\n","MES = set(MES)"]},{"cell_type":"markdown","metadata":{},"source":["### Add Early Endodermal lineage markers \n","https://www.rndsystems.com/research-area/early-endodermal-lineage-markers"]},{"cell_type":"code","execution_count":6,"metadata":{},"outputs":[],"source":["# Definitive Endoderm Markers\n","DE = [\n","    \"Cldn6\",  # Claudin-6\n","    \"Krt19\",  # Cytokeratin 19\n","    \"Eomes\",  # EOMES\n","    \"Fabp1\",  # FABP1/L-FABP\n","    \"Fabp2\",  # FABP2/I-FABP\n","    \"Gata4\",  # GATA-4\n","    \"Gsc\",    # Goosecoid\n","    \"Foxa1\",  # FoxA1/HNF-3 alpha\n","    \"Foxa2\",  # FoxA2/HNF-3 beta\n","    \"Sox7\",   # SOX7\n","    \"Sox17\",  # SOX17\n","    \"Hnf1b\"   # HNF-1 beta/TCF-2\n","]\n","\n","# Primitive Endoderm Markers\n","PrE = [\n","    \"Afp\",    # alpha-Fetoprotein/AFP\n","    \"Ctnnb1\", # beta-Catenin\n","    \"Gata4\",  # GATA-4\n","    \"Gata6\",  # GATA-6\n","    \"Gdf1\",   # GDF-1\n","    \"Gdf3\",   # GDF-3\n","    \"Hnf4a\",  # HNF-4 alpha/NR2A1\n","    \"Mixl1\",  # MIXL1\n","    \"Sall4\",  # SALL4\n","    \"Sox7\",   # SOX7\n","    \"Sox17\"   # SOX17\n","]\n","PrE = set(PrE)\n","DE = set(DE)"]},{"cell_type":"code","execution_count":7,"metadata":{},"outputs":[],"source":["mouse_genes = pd.read_csv('../data/matrices/RNA_FPKM2500.csv')['GENE']\n","mouse_genes = set(mouse_genes)\n","assert len(MES.difference(mouse_genes))==0\n","assert len(PrE.difference(mouse_genes))==0\n","assert len(DE.difference(mouse_genes))==0"]},{"cell_type":"code","execution_count":8,"metadata":{},"outputs":[],"source":["GENE_SETS = {  \n","                'Hedgehog Signaling Pathway WP116':wp['Hedgehog Signaling Pathway WP116'],\n","                #'PluriNetWork WP1763',\n","                'ESC Pluripotency Pathways WP339':wp['ESC Pluripotency Pathways WP339'],\n","                'Wnt Signaling Pathway WP403': wp['Wnt Signaling Pathway WP403'],\n","                'Early Mesodermal Lineage Markers': MES,\n","                'Heart Development WP2067': wp['Heart Development WP2067'],\n","                'Delta-Notch Signaling Pathway WP265': wp['Delta-Notch Signaling Pathway WP265'],\n","                'Neural Crest Differentiation WP2074': wp['Neural Crest Differentiation WP2074'],\n","                'Striated Muscle Contraction WP216': wp['Striated Muscle Contraction WP216'], \n","                \n","                'Definitive Endoderm Markers': DE,\n","                }"]},{"cell_type":"markdown","metadata":{},"source":["### Map Gene names Human2Mouse (Manually AAA123A-> Aaa123a)"]},{"cell_type":"code","execution_count":9,"metadata":{},"outputs":[],"source":["def hum2mouse(gene):\n","\tnew_gene = gene[0] + gene[1:].lower()\n","\treturn new_gene\n"]},{"cell_type":"code","execution_count":10,"metadata":{},"outputs":[],"source":["GENE_SETS_low = {}\n","\n","for term, genes in GENE_SETS.items():\n","    new_genes = []\n","    for gene in genes: \n","            new_gene = hum2mouse(gene)\n","            new_genes.append(new_gene)\n","    GENE_SETS_low[term] = new_genes"]},{"cell_type":"code","execution_count":11,"metadata":{},"outputs":[],"source":["! mkdir -p ../data/gene_sets\n","with open(f'../data/gene_sets/gene_sets_dict.pkl', 'wb') as f:\n","        pickle.dump(GENE_SETS_low, f)"]},{"cell_type":"markdown","metadata":{},"source":["# Bivalent/Active promoters ESC Gonzalez"]},{"cell_type":"code","execution_count":12,"metadata":{},"outputs":[],"source":["ACT = set(pd.read_excel('../../00_RegionAnnotation/Gonzalez/active_promoters.xlsx')['gene'])\n","BIV = set(pd.read_excel('../../00_RegionAnnotation/Gonzalez/bivalent_promoters.xlsx')['gene'])\n","GONZALEZ = {  'Active':ACT, \n","            'Bivalent':BIV}\n","with open(f'../data/gene_sets/gonzalez_dict.pkl', 'wb') as f:\n","        pickle.dump(GONZALEZ, f)"]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyMDC1ToWGIBGccWWiHT+eFl","provenance":[{"file_id":"1B-9-kc0cCv3P5WHf7rDrocRrN5fxLnTK","timestamp":1622642792385}]},"kernelspec":{"display_name":"gseapy","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.14"}},"nbformat":4,"nbformat_minor":0}
