{"cells":[{"cell_type":"markdown","metadata":{"id":"lCcAA73VDWoi"},"source":["### Set parameters and names"]},{"cell_type":"code","execution_count":1,"metadata":{},"outputs":[],"source":["import pandas as pd\n","import matplotlib as mpl\n","import matplotlib.pyplot as plt\n","\n","import gseapy as gp\n","from gseapy import barplot, dotplot\n","import pickle        \n"]},{"cell_type":"code","execution_count":2,"metadata":{},"outputs":[],"source":["mpl.rc('axes', labelsize=14)\n","mpl.rc('xtick', labelsize=12)\n","mpl.rc('ytick', labelsize=12)\n","plt.rcParams['savefig.dpi'] = 250\n","plt.rcParams['figure.dpi'] = 100\n","plt.rcParams['axes.facecolor'] = 'white'\n","%matplotlib inline"]},{"cell_type":"code","execution_count":3,"metadata":{},"outputs":[],"source":["INPUT_GENES    = 'ALL'\n","INPUT_FEATURES = 'X_FC'\n","INPUT_NORM     = 'z'\n","CODINGS_SIZE = 6\n","\n","ID     = f'{CODINGS_SIZE}D_{INPUT_GENES}_{INPUT_FEATURES}_{INPUT_NORM}'\n","METHOD = 'VAE'\n","k      = 80\n","LABELS_COL = f'GMM_{METHOD}_{k}'\n","DIR_DATA= f'../data/{ID}_analysis/{LABELS_COL}/'\n","DIR_FIG = f'../figures/{ID}_analysis/{METHOD}/ORA_k{k}'\n","! mkdir -p {DIR_FIG}\n","\n"]},{"cell_type":"code","execution_count":4,"metadata":{},"outputs":[],"source":["with open(f'{DIR_DATA}gene_clusters_dict.pkl', 'rb') as f:\n","    GENE_CLUSTERS = pickle.load(f)\n"]},{"cell_type":"code","execution_count":5,"metadata":{},"outputs":[{"data":{"text/plain":["['ARCHS4_Cell-lines',\n"," 'ARCHS4_IDG_Coexp',\n"," 'ARCHS4_Kinases_Coexp',\n"," 'ARCHS4_TFs_Coexp',\n"," 'ARCHS4_Tissues',\n"," 'Achilles_fitness_decrease',\n"," 'Achilles_fitness_increase',\n"," 'Aging_Perturbations_from_GEO_down',\n"," 'Aging_Perturbations_from_GEO_up',\n"," 'Allen_Brain_Atlas_10x_scRNA_2021',\n"," 'Allen_Brain_Atlas_down',\n"," 'Allen_Brain_Atlas_up',\n"," 'Azimuth_2023',\n"," 'Azimuth_Cell_Types_2021',\n"," 'BioCarta_2013',\n"," 'BioCarta_2015',\n"," 'BioCarta_2016',\n"," 'BioPlanet_2019',\n"," 'BioPlex_2017',\n"," 'CCLE_Proteomics_2020',\n"," 'CORUM',\n"," 'COVID-19_Related_Gene_Sets',\n"," 'COVID-19_Related_Gene_Sets_2021',\n"," 'Cancer_Cell_Line_Encyclopedia',\n"," 'CellMarker_2024',\n"," 'CellMarker_Augmented_2021',\n"," 'ChEA_2013',\n"," 'ChEA_2015',\n"," 'ChEA_2016',\n"," 'ChEA_2022',\n"," 'Chromosome_Location',\n"," 'Chromosome_Location_hg19',\n"," 'ClinVar_2019',\n"," 'DSigDB',\n"," 'Data_Acquisition_Method_Most_Popular_Genes',\n"," 'DepMap_WG_CRISPR_Screens_Broad_CellLines_2019',\n"," 'DepMap_WG_CRISPR_Screens_Sanger_CellLines_2019',\n"," 'Descartes_Cell_Types_and_Tissue_2021',\n"," 'Diabetes_Perturbations_GEO_2022',\n"," 'DisGeNET',\n"," 'Disease_Perturbations_from_GEO_down',\n"," 'Disease_Perturbations_from_GEO_up',\n"," 'Disease_Signatures_from_GEO_down_2014',\n"," 'Disease_Signatures_from_GEO_up_2014',\n"," 'DrugMatrix',\n"," 'Drug_Perturbations_from_GEO_2014',\n"," 'Drug_Perturbations_from_GEO_down',\n"," 'Drug_Perturbations_from_GEO_up',\n"," 'ENCODE_Histone_Modifications_2013',\n"," 'ENCODE_Histone_Modifications_2015',\n"," 'ENCODE_TF_ChIP-seq_2014',\n"," 'ENCODE_TF_ChIP-seq_2015',\n"," 'ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X',\n"," 'ESCAPE',\n"," 'Elsevier_Pathway_Collection',\n"," 'Enrichr_Libraries_Most_Popular_Genes',\n"," 'Enrichr_Submissions_TF-Gene_Coocurrence',\n"," 'Enrichr_Users_Contributed_Lists_2020',\n"," 'Epigenomics_Roadmap_HM_ChIP-seq',\n"," 'FANTOM6_lncRNA_KD_DEGs',\n"," 'GO_Biological_Process_2013',\n"," 'GO_Biological_Process_2015',\n"," 'GO_Biological_Process_2017',\n"," 'GO_Biological_Process_2017b',\n"," 'GO_Biological_Process_2018',\n"," 'GO_Biological_Process_2021',\n"," 'GO_Biological_Process_2023',\n"," 'GO_Cellular_Component_2013',\n"," 'GO_Cellular_Component_2015',\n"," 'GO_Cellular_Component_2017',\n"," 'GO_Cellular_Component_2017b',\n"," 'GO_Cellular_Component_2018',\n"," 'GO_Cellular_Component_2021',\n"," 'GO_Cellular_Component_2023',\n"," 'GO_Molecular_Function_2013',\n"," 'GO_Molecular_Function_2015',\n"," 'GO_Molecular_Function_2017',\n"," 'GO_Molecular_Function_2017b',\n"," 'GO_Molecular_Function_2018',\n"," 'GO_Molecular_Function_2021',\n"," 'GO_Molecular_Function_2023',\n"," 'GTEx_Aging_Signatures_2021',\n"," 'GTEx_Tissue_Expression_Down',\n"," 'GTEx_Tissue_Expression_Up',\n"," 'GTEx_Tissues_V8_2023',\n"," 'GWAS_Catalog_2019',\n"," 'GWAS_Catalog_2023',\n"," 'GeDiPNet_2023',\n"," 'GeneSigDB',\n"," 'Gene_Perturbations_from_GEO_down',\n"," 'Gene_Perturbations_from_GEO_up',\n"," 'Genes_Associated_with_NIH_Grants',\n"," 'Genome_Browser_PWMs',\n"," 'GlyGen_Glycosylated_Proteins_2022',\n"," 'HDSigDB_Human_2021',\n"," 'HDSigDB_Mouse_2021',\n"," 'HMDB_Metabolites',\n"," 'HMS_LINCS_KinomeScan',\n"," 'HomoloGene',\n"," 'HuBMAP_ASCT_plus_B_augmented_w_RNAseq_Coexpression',\n"," 'HuBMAP_ASCTplusB_augmented_2022',\n"," 'HumanCyc_2015',\n"," 'HumanCyc_2016',\n"," 'Human_Gene_Atlas',\n"," 'Human_Phenotype_Ontology',\n"," 'IDG_Drug_Targets_2022',\n"," 'InterPro_Domains_2019',\n"," 'Jensen_COMPARTMENTS',\n"," 'Jensen_DISEASES',\n"," 'Jensen_TISSUES',\n"," 'KEA_2013',\n"," 'KEA_2015',\n"," 'KEGG_2013',\n"," 'KEGG_2015',\n"," 'KEGG_2016',\n"," 'KEGG_2019_Human',\n"," 'KEGG_2019_Mouse',\n"," 'KEGG_2021_Human',\n"," 'KOMP2_Mouse_Phenotypes_2022',\n"," 'Kinase_Perturbations_from_GEO_down',\n"," 'Kinase_Perturbations_from_GEO_up',\n"," 'L1000_Kinase_and_GPCR_Perturbations_down',\n"," 'L1000_Kinase_and_GPCR_Perturbations_up',\n"," 'LINCS_L1000_CRISPR_KO_Consensus_Sigs',\n"," 'LINCS_L1000_Chem_Pert_Consensus_Sigs',\n"," 'LINCS_L1000_Chem_Pert_down',\n"," 'LINCS_L1000_Chem_Pert_up',\n"," 'LINCS_L1000_Ligand_Perturbations_down',\n"," 'LINCS_L1000_Ligand_Perturbations_up',\n"," 'Ligand_Perturbations_from_GEO_down',\n"," 'Ligand_Perturbations_from_GEO_up',\n"," 'MAGMA_Drugs_and_Diseases',\n"," 'MAGNET_2023',\n"," 'MCF7_Perturbations_from_GEO_down',\n"," 'MCF7_Perturbations_from_GEO_up',\n"," 'MGI_Mammalian_Phenotype_2013',\n"," 'MGI_Mammalian_Phenotype_2017',\n"," 'MGI_Mammalian_Phenotype_Level_3',\n"," 'MGI_Mammalian_Phenotype_Level_4',\n"," 'MGI_Mammalian_Phenotype_Level_4_2019',\n"," 'MGI_Mammalian_Phenotype_Level_4_2021',\n"," 'MSigDB_Computational',\n"," 'MSigDB_Hallmark_2020',\n"," 'MSigDB_Oncogenic_Signatures',\n"," 'Metabolomics_Workbench_Metabolites_2022',\n"," 'Microbe_Perturbations_from_GEO_down',\n"," 'Microbe_Perturbations_from_GEO_up',\n"," 'MoTrPAC_2023',\n"," 'Mouse_Gene_Atlas',\n"," 'NCI-60_Cancer_Cell_Lines',\n"," 'NCI-Nature_2015',\n"," 'NCI-Nature_2016',\n"," 'NIH_Funded_PIs_2017_AutoRIF_ARCHS4_Predictions',\n"," 'NIH_Funded_PIs_2017_GeneRIF_ARCHS4_Predictions',\n"," 'NIH_Funded_PIs_2017_Human_AutoRIF',\n"," 'NIH_Funded_PIs_2017_Human_GeneRIF',\n"," 'NURSA_Human_Endogenous_Complexome',\n"," 'OMIM_Disease',\n"," 'OMIM_Expanded',\n"," 'Old_CMAP_down',\n"," 'Old_CMAP_up',\n"," 'Orphanet_Augmented_2021',\n"," 'PFOCR_Pathways',\n"," 'PFOCR_Pathways_2023',\n"," 'PPI_Hub_Proteins',\n"," 'PanglaoDB_Augmented_2021',\n"," 'Panther_2015',\n"," 'Panther_2016',\n"," 'Pfam_Domains_2019',\n"," 'Pfam_InterPro_Domains',\n"," 'PheWeb_2019',\n"," 'PhenGenI_Association_2021',\n"," 'Phosphatase_Substrates_from_DEPOD',\n"," 'ProteomicsDB_2020',\n"," 'Proteomics_Drug_Atlas_2023',\n"," 'RNA-Seq_Disease_Gene_and_Drug_Signatures_from_GEO',\n"," 'RNAseq_Automatic_GEO_Signatures_Human_Down',\n"," 'RNAseq_Automatic_GEO_Signatures_Human_Up',\n"," 'RNAseq_Automatic_GEO_Signatures_Mouse_Down',\n"," 'RNAseq_Automatic_GEO_Signatures_Mouse_Up',\n"," 'Rare_Diseases_AutoRIF_ARCHS4_Predictions',\n"," 'Rare_Diseases_AutoRIF_Gene_Lists',\n"," 'Rare_Diseases_GeneRIF_ARCHS4_Predictions',\n"," 'Rare_Diseases_GeneRIF_Gene_Lists',\n"," 'Reactome_2013',\n"," 'Reactome_2015',\n"," 'Reactome_2016',\n"," 'Reactome_2022',\n"," 'Rummagene_kinases',\n"," 'Rummagene_signatures',\n"," 'Rummagene_transcription_factors',\n"," 'SILAC_Phosphoproteomics',\n"," 'SubCell_BarCode',\n"," 'SynGO_2022',\n"," 'SynGO_2024',\n"," 'SysMyo_Muscle_Gene_Sets',\n"," 'TF-LOF_Expression_from_GEO',\n"," 'TF_Perturbations_Followed_by_Expression',\n"," 'TG_GATES_2020',\n"," 'TRANSFAC_and_JASPAR_PWMs',\n"," 'TRRUST_Transcription_Factors_2019',\n"," 'Table_Mining_of_CRISPR_Studies',\n"," 'Tabula_Muris',\n"," 'Tabula_Sapiens',\n"," 'TargetScan_microRNA',\n"," 'TargetScan_microRNA_2017',\n"," 'The_Kinase_Library_2023',\n"," 'Tissue_Protein_Expression_from_Human_Proteome_Map',\n"," 'Tissue_Protein_Expression_from_ProteomicsDB',\n"," 'Transcription_Factor_PPIs',\n"," 'UK_Biobank_GWAS_v1',\n"," 'Virus-Host_PPI_P-HIPSTer_2020',\n"," 'VirusMINT',\n"," 'Virus_Perturbations_from_GEO_down',\n"," 'Virus_Perturbations_from_GEO_up',\n"," 'WikiPathway_2021_Human',\n"," 'WikiPathway_2023_Human',\n"," 'WikiPathways_2013',\n"," 'WikiPathways_2015',\n"," 'WikiPathways_2016',\n"," 'WikiPathways_2019_Human',\n"," 'WikiPathways_2019_Mouse',\n"," 'dbGaP',\n"," 'huMAP',\n"," 'lncHUB_lncRNA_Co-Expression',\n"," 'miRTarBase_2017']"]},"execution_count":5,"metadata":{},"output_type":"execute_result"}],"source":["\n","MOUSE_DB = gp.get_library_name(organism='Mouse')\n","MOUSE_DB\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["RESULTS = {}\n","for k, value in GENE_CLUSTERS.items():\n","    enr = gp.enrichr(gene_list= value['gene_list'],\n","                    background=  f'../data/{ID}_analysis/background.list',\n","                    gene_sets=['KEGG_2019_Mouse','WikiPathways_2019_Mouse','GO_Biological_Process_2023','GO_Molecular_Function_2023'],\n","                    organism='Mouse', \n","                    outdir=None, \n","                    verbose=True )\n","    \n","    RESULTS[f'Cluster_{k}']= {'result': enr.results, 'n_genes':value['len']}"]},{"cell_type":"markdown","metadata":{},"source":["### Remove from Term final string after '('"]},{"cell_type":"code","execution_count":7,"metadata":{},"outputs":[],"source":["RESULTS_CLEANED = RESULTS.copy()\n","for key,value in RESULTS_CLEANED.items():\n","    x=RESULTS_CLEANED[key]['result']['Term'].str\n","    NEW_COL=[]\n","    for el in list(x.split('(')):\n","        NEW_COL.append(el[0])\n","    RESULTS_CLEANED[key]['result']['Term']=NEW_COL"]},{"cell_type":"code","execution_count":8,"metadata":{},"outputs":[],"source":["COL_DICT = {\n","    'KEGG_2019_Mouse': '#66C2A5',\n","    'WikiPathways_2019_Mouse': '#FC8D62',\n","    'GO_Biological_Process_2023': '#8DA0CB',\n","    'GO_Molecular_Function_2023': '#E78AC3'\n","}\n"]},{"cell_type":"code","execution_count":null,"metadata":{},"outputs":[],"source":["for cluster_k, value in RESULTS_CLEANED.items():\n","        n_genes=RESULTS_CLEANED[cluster_k]['n_genes']\n","        \n","        df = RESULTS_CLEANED[cluster_k]['result']\n","                #\n","        try:\n","                barplot(df,\n","                        column=\"Adjusted P-value\",\n","                        group='Gene_set',\n","                        color=COL_DICT,\n","                        top_term=5,\n","                        title=f'{cluster_k}, n={n_genes}',\n","                        #xticklabels_rot=45,\n","                        #show_ring=False,\n","                        #marker='o',\n","                        #cmap='viridis'\n","                        )\n","                plt.savefig(f'{DIR_FIG}/{cluster_k}.pdf', format=\"pdf\", bbox_inches=\"tight\")\n","        except ValueError:\n","                with open(f'{DIR_FIG}/{cluster_k}_no_sign_terms.error', \"w\") as file:\n","                        file.write(\"No significant terms\")                    \n","        if cluster_k=='5':break\n","# Adjust layout\n","plt.tight_layout()\n","# Show the plot\n","plt.show()\n"]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyMDC1ToWGIBGccWWiHT+eFl","provenance":[{"file_id":"1B-9-kc0cCv3P5WHf7rDrocRrN5fxLnTK","timestamp":1622642792385}]},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.14"}},"nbformat":4,"nbformat_minor":0}
